File: app/src/main/java/com/example/appestable/domain/SessionContext.kt
package com.example.appestable.domain

import com.example.appestable.data.RolViaje

data class SessionContext(
    val viajeId: Int,
    val viajeNombre: String,
    val rol: RolViaje,
    val familiaId: Int? = null,
    val usuarioId: Int? = null,
    val isLoggedIn: Boolean = false
) {
    companion object {
        val GUEST_ORGANIZER = SessionContext(
            viajeId = 1,
            viajeNombre = "Viaje Principal",
            rol = RolViaje.ORGANIZADOR,
            isLoggedIn = false
        )
    }
}

File: flutter_app/lib/domain/session_context.dart
import '../models/rol_viaje.dart';

class SessionContext {
  final int viajeId;
  final String viajeNombre;
  final RolViaje rol;
  final int? familiaId;
  final int? usuarioId;
  final bool isLoggedIn;

  const SessionContext({
    required this.viajeId,
    required this.viajeNombre,
    required this.rol,
    this.familiaId,
    this.usuarioId,
    this.isLoggedIn = false,
  });

  static const SessionContext empty = SessionContext(
    viajeId: 0,
    viajeNombre: '',
    rol: RolViaje.miembro,
    isLoggedIn: false,
  );
}